polaris-icons/rollup.config.mjs

```diff
@@ -55,15 +55,18 @@ const metadataContent = `
 const metadata = ${JSON.stringify(iconMetadata, null, 2)};
 export default metadata;
 `.trim();
-const metadataTypes = `declare const metadata: {
-  [key: string]: {
-    id: string;
-    name: string;
-    set: 'major' | 'minor';
-    description: string;
-    keywords: string[];
-  };
+const metadataTypes = `export interface Icon {
+  id: string;
+  name: string;
+  set: 'major' | 'minor';
+  description: string;
+  keywords: string[];
+}
+
+declare const metadata: {
+  [key: string]: Icon;
```

Comment:
Mind adding a meaningful name to the `index signature` key? e.g.
`[icon: string]`, `[name: string]`, `[iconName: string]`
```ts
declare const metadata: {
  [iconName: string]: Icon;
}
```

Reply:
Thanks @aaronccasanova brought this into https://github.com/Shopify/polaris/pull/6439